### PR TITLE
Basic auth to fetch Avro schema in Kafka

### PR DESCRIPTION
**Pull request by ilejn**
### Changelog category (leave one):
- New Feature 

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Kafka connector can fetch avro schema from schema registry with basic authentication using url-encoded credentials

### CODE REVIEW

**Comment by ilejn** [comment]
Based on https://github.com/ClickHouse/ClickHouse/pull/38948

**Comment by robot-ch-test-poll** [comment]
This is an automated comment for commit 1f8733afece39d5e8ac267922371f275c2924e6a with description of existing statuses. It's updated for the latest CI running
The full report is available [here](https://s3.amazonaws.com/clickhouse-test-reports/49664/1f8733afece39d5e8ac267922371f275c2924e6a/ci_running.html)
The overall status of the commit is 🟡 pending

<table><thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody><tr><td>CI running</td><td>A meta-check that indicates the running CI. Normally, it's in <b>success</b> or <b>pending</b> state. The failed status indicates some problems with the PR</td><td>🟡 pending</td></tr>
<tr><td>Docs Check</td><td>Builds and tests the documentation</td><td>🟡 pending</td></tr>
</table>

**Comment by ilejn** [comment]
Issue (with window_view_alter_query_watch) seems not related.

**Comment by lesandie** [comment]
@ilejn if you need help testing say so!

**Comment by ilejn** [comment]
> @ilejn if you need help testing say so!

Well, testing a test sounds kinda strange, although why not?
It might be more useful to test the functionality itself, while I cannot imagine any security risk here.

File: tests/integration/test_format_avro_confluent/test.py [hunk 1]
```diff
@@ -75,3 +84,110 @@ def test_select(started_cluster):
...
+        message = serializer.encode_record_with_schema(
+            "test_subject_auth_encoded", schema, {"value": x}
+        )
+        buf.write(message)
+    data = buf.getvalue()
+
+    instance = started_cluster.instances["dummy"]  # type: ClickHouseInstance
+    schema_registry_url = "http://{}:{}@{}:{}".format(
+        parse.quote_plus("schemauser/slash"),
+        parse.quote_plus("letmein"),
```
[the remark this answers is not in the document]
Comment (ilejn) [reply tests/integration/test_format_avro_confluent/test.py:171]: Done as a new test

File: tests/integration/test_format_avro_confluent/test.py [hunk 2]
```diff
@@ -75,3 +84,110 @@ def test_select(started_cluster):
         ["1"],
         ["2"],
     ]
+
+
+def test_select_auth(started_cluster):
+    # type: (ClickHouseCluster) -> None
+    time.sleep(5)
```
[the remark this answers is not in the document]
Comment (ilejn) [reply tests/integration/test_format_avro_confluent/test.py]: It is not needed ;)
Thanks.

**Comment by ilejn** [comment]
I've examined logs, almost sure that test issues are not related.

**Comment by ilejn** [comment]
Could someone please review this PR?

**Comment by ilejn** [comment]
@evillique , could you have a look? This is basically a test for https://github.com/ClickHouse/ClickHouse/pull/38948

File: tests/integration/test_format_avro_confluent/test.py [hunk 3]
```diff
@@ -36,18 +38,25 @@ def run_query(instance, query, data=None, settings=None):
 
     return result
 
+    # reg_url="http://localhost:{}".format(started_cluster.schema_registry_port)
+    # arg={'url':reg_url}
+    # schema_registry_client = CachedSchemaRegistryClient(arg)
+
```
Comment (evillique) [inline tests/integration/test_format_avro_confluent/test.py:44]: I suppose this is not needed
```suggestion
```

Comment (ilejn) [reply tests/integration/test_format_avro_confluent/test.py]: Ok. Removed.

File: tests/integration/test_format_avro_confluent/test.py [hunk 4]
```diff
@@ -36,18 +38,25 @@ def run_query(instance, query, data=None, settings=None):
...
 
 def test_select(started_cluster):
     # type: (ClickHouseCluster) -> None
 
-    schema_registry_client = CachedSchemaRegistryClient(
-        "http://localhost:{}".format(started_cluster.schema_registry_port)
-    )
+    # input("Top of test_select, press any key")
+
```
[the remark this answers is not in the document]
Comment (ilejn) [reply tests/integration/test_format_avro_confluent/test.py]: Ok. Removed.

**Comment by ilejn** [comment]
Attn @antaljanosbenjamin

**Review by evillique — APPROVED** [review]

**Comment by ilejn** [comment]
Hello @evillique , should I rebase to fresh master or what ?

**Comment by evillique** [comment]
@ilejn Yes, that would be helpful, tests are failing for unrelated reasons.

**Comment by ilejn** [comment]
> @ilejn Yes, that would be helpful, tests are failing for unrelated reasons.

Ok.
Was not sure that it worth burning time/CPU/money on another CI run.

**Pull request merged**